<!--
 Copyright 2022 Shadow Robot Company Ltd.
 This program is free software: you can redistribute it and/or modify it
 under the terms of the GNU General Public License as published by the Free
 Software Foundation version 2 of the License.
 This program is distributed in the hope that it will be useful, but WITHOUT
 ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
 more details.
 You should have received a copy of the GNU General Public License along
 with this program. If not, see <http://www.gnu.org/licenses/>.
-->

<launch>
    <arg name="package_name" default="sr_multi_moveit_config" />
    <arg name="executable_name" default="scene.launch" />
    <arg name="topics_list" default="[]" />
    <arg name="params_list" default="[/robot_description]" />
    <arg name="services_list" default="[]" />
    <arg name="scene_file" default=""/>
    <arg name="launch_args_list" default=""/>
    <arg name="timeout" default="3.0"/>

    <node name="conditional_delayed_rostool" pkg="sr_utilities_common" type="conditional_delayed_rostool.py" output="screen">
        <param name="package_name" value="$(arg package_name)" />
        <param name="executable_name" value="$(arg executable_name)" />
        <rosparam param="topics_list" subst_value="True">$(arg topics_list)</rosparam>
        <rosparam param="params_list" subst_value="True">$(arg params_list)</rosparam>
        <rosparam param="services_list" subst_value="True">$(arg services_list)</rosparam>
        <param name="launch_args_list" value="scene_file:=$(arg scene_file)" />
        <param name="timeout" value="$(arg timeout)" />
    </node>
</launch>
